ihmc-robotics-toolkit/src/main/java/us/ihmc/robotics/math/filters/TransferFunctionDiscretizer.java

```diff
@@ -0,0 +1,204 @@
+
+   /*
+    * @param name Name of the filter.
+    * @param k Transfer Function Gain
+    * @param numerator Order of continuous polynomial coefficients from largest to smallest order, i.e. [b_0, b_1, ... , b_{m-1}, b_{m}].
+    * @param demoninator Order of continuous polynomial coefficients from largest to smallest order, i.e. [a_0, a_1, ..., a_{n-1}, a_n].
+    * @param sampleFrequency Frequency [Hz] which this variable will be updated. Filter performance will be worse if this is not accurate.
+    */
+   public TransferFunctionDiscretizer(String name, ContinuousTransferFunction tf, double sampleFrequency)
+   {
+
+      this.name = name;
+
+      m = tf.getNumerator().length;
+      n = tf.getDenominator().length;
+
+      // Handles multiple cases of m and n for matrix sizes.
+      if (m > n)
+      {
+         throw new IllegalArgumentException("Transfer Function is non-causal!");
+      }
+      else if (m < n)
+      {
+         double[] d = new double[n];
+         System.arraycopy(tf.getNumerator(), 0, d, n - m, m);
+         this.numerator = new DMatrixRMaj(1, d.length, false, d);
+      }
+      else
+      {
+         this.numerator = new DMatrixRMaj(1, m, false, tf.getNumerator());
+      }
+
+      this.denominator = new DMatrixRMaj(1, n, false, tf.getDenominator());
+
+      this.k = tf.getGain();
+      this.fs = sampleFrequency;
+      this.p = 1 / (2 * this.fs);
+
+      // Apply gain across numerator.
+      CommonOps_DDRM.scale(k, this.numerator);
+
+      // Solve for the corresponding input/output filter coefficients.
+      buildFilter();
+   }
+
+   private void buildFilter()
+   {
+
+      numerator = solveFx_to_Fz(numerator);
+      denominator = solveFx_to_Fz(denominator);
+
+      // Solve for outputCoefficients.
+      inputCoefficients = numerator.copy();
+      CommonOps_DDRM.scale(1.0 / denominator.get(0), inputCoefficients);
+
+      // Solve for inputCoefficients.
+      double[] d = new double[n - 1];
+      System.arraycopy(denominator.data, 1, d, 0, denominator.numCols - 1);
+      outputCoefficients = new DMatrixRMaj(1, d.length, false, d);
+      CommonOps_DDRM.scale(-1.0 / denominator.get(0), outputCoefficients);
+   }
+
+   private double calc;
+
+   private DMatrixRMaj solveFx_to_Fz(DMatrixRMaj Fx)
+   {
+      numCols = Fx.numCols;
+      DMatrixRMaj Fz = new DMatrixRMaj(1, numCols);
+
+      /*
+       *  1) 	Dividing by s^n. Multiply out the correct number of gains.
+       *  	Converting Numerator & Denominator to F(x) form, where they
+       *  	are backwards in this form.
+       */
+      calc = 1 / p;
+      for (int i = 0; i < n; i++)
+      {
+         calc *= p;
+         Fx.times(i, calc);
+      }
+      Fx = reverseVector(Fx);
+
+      // 2) Apply Synthetic Division. F(x) -> F(x+1).
+      Fx = syntheticDivision(Fx, 1.0);
+
+      // 3)  Flip Coefficients. F(x+1) -> F(1/x + 1).
+      Fx = reverseVector(Fx);
+
+      // 4) Convert F(1/x + 1) -> F(2/x + 1).
+      calc = 1.0 / 2.0;

```

Comment:
why would you write it this way, as opposed to 0.5? This does an actual operation, while writing 0.5 doesn't.